src/index.js

```diff
@@ -1,32 +1,43 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+
+function fetchData(url_api) {
+  return new Promise( (resolv, reject)=>{
+    let xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange =  (event) => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200)          
+          resolv( JSON.parse( xhttp.responseText) );
+        else 
+          return reject(url_api);
+      }
+    };
+    xhttp.open('GET', url_api, true);
+    xhttp.send();
+  })
 };
 
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
+
+fetchData( API )
+  .then( (data) =>{
+    console.log('Primer Llamado...')
+    let url = API + data.results[0].id;
```

Comment:
No olvidemos seguir utilizando template literals
```suggestion
    let url = `${API}${data.results[0].id}`;
```